drivers/pwm/pwm_stm32.c

```diff
@@ -26,27 +26,6 @@
 
 #define CHANNEL_LENGTH 4
 
-static u32_t __get_tim_clk(u32_t bus_clk,
-			      clock_control_subsys_t *sub_system)
-{
-	struct stm32_pclken *pclken = (struct stm32_pclken *)(sub_system);
-	u32_t tim_clk, apb_psc;
-
-	if (pclken->bus == STM32_CLOCK_BUS_APB1) {
-		apb_psc = CONFIG_CLOCK_STM32_APB1_PRESCALER;
-	} else {
-		apb_psc = CONFIG_CLOCK_STM32_APB2_PRESCALER;
-	}
-
-	if (apb_psc == RCC_HCLK_DIV1) {
```

Comment:
There is a bug here. The RCC_HCLK_DIV1 macro will resolve to 0x00 (value of the PPRE1/2 register). But on the other side, CONFIG_CLOCK_STM32_APB[12]_PRESCALER will use a value of 1 (or more) if we want a prescaler of one (or more). Thus the condition is always false. RCC_HCLK_DIV1 must be replaced by the literal "0x01" instead or a similar macro.

I have a patch for that in my tree, but had no time to upstream the changes. So feel free.